src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs

```diff
@@ -152,5 +152,134 @@ public static void GetHRForException()
                 Marshal.GetExceptionForHR(e.HResult);
             }
         }
+
+        [Fact]
+        public static void GenerateGuidForType()
+        {
+             Assert.Throws<PlatformNotSupportedException>(() => Marshal.GenerateGuidForType(typeof(System.Int32)));        
```

Comment:
@stephentoub OK. Returning Type.Guid seems fine. That is also what Mono is doing.